pennylane/qchem/convert.py

```diff
@@ -354,3 +355,264 @@ def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e01
+    >>> wf_cisd = cisd_state(myci, tol=1e-1)
+    >>> print(wf_cisd)
+    {(1, 1): -0.9942969785398778, (2, 2): 0.10664669927602159}
+    """
+    mol = cisd_solver.mol
+    cisdvec = cisd_solver.ci
+
+    norb = mol.nao
+    nelec = mol.nelectron
+    nelec_a = int((nelec + mol.spin) / 2)
+    nelec_b = int((nelec - mol.spin) / 2)
+
+    nvir_a, nvir_b = norb - nelec_a, norb - nelec_b
+
+    size_a, size_b = nelec_a * nvir_a, nelec_b * nvir_b
+    size_aa = int(nelec_a * (nelec_a - 1) / 2) * int(nvir_a * (nvir_a - 1) / 2)
+    size_bb = int(nelec_b * (nelec_b - 1) / 2) * int(nvir_b * (nvir_b - 1) / 2)
+    size_ab = nelec_a * nelec_b * nvir_a * nvir_b
+
+    sizes = [1, size_a, size_b, size_aa, size_ab, size_bb]
+    cumul = np.cumsum(sizes)
+    idxs = [0] + [slice(cumul[ii], cumul[ii + 1]) for ii in range(len(cumul) - 1)]
+    c0, c1a, c1b, c2aa, c2ab, c2bb = [cisdvec[idx] for idx in idxs]
+
+    # numbers representing the Hartree-Fock vector, e.g., bin(ref_a)[::-1] = 1111...10...0
+    ref_a = int(2**nelec_a - 1)
+    ref_b = int(2**nelec_b - 1)
+
+    dict_fcimatr = dict(zip(list(zip([ref_a], [ref_b])), [c0]))
+
+    # alpha -> alpha excitations
+    c1a_configs, c1a_signs = _excited_configurations(nelec_a, norb, 1)
+    dict_fcimatr.update(dict(zip(list(zip(c1a_configs, [ref_b] * size_a)), c1a * c1a_signs)))
+
+    # beta -> beta excitations
+    c1b_configs, c1b_signs = _excited_configurations(nelec_b, norb, 1)
+    dict_fcimatr.update(dict(zip(list(zip([ref_a] * size_b, c1b_configs)), c1b * c1b_signs)))
+
+    # alpha, alpha -> alpha, alpha excitations
+    c2aa_configs, c2aa_signs = _excited_configurations(nelec_a, norb, 2)
+    dict_fcimatr.update(dict(zip(list(zip(c2aa_configs, [ref_b] * size_aa)), c2aa * c2aa_signs)))
+
+    # alpha, beta -> alpha, beta excitations
+    rowvals, colvals = np.array(list(product(c1a_configs, c1b_configs)), dtype=int).T.numpy()
+    dict_fcimatr.update(
+        dict(zip(list(zip(rowvals, colvals)), c2ab * np.kron(c1a_signs, c1b_signs).numpy()))
+    )
+
+    # beta, beta -> beta, beta excitations
+    c2bb_configs, c2bb_signs = _excited_configurations(nelec_b, norb, 2)
+    dict_fcimatr.update(dict(zip(list(zip([ref_a] * size_bb, c2bb_configs)), c2bb * c2bb_signs)))
+
+    # filter based on tolerance cutoff
+    dict_fcimatr = {key: value for key, value in dict_fcimatr.items() if abs(value) > tol}
+
+    return dict_fcimatr
+
+
+def import_state(solver, method, tol=1e-15):
+    r"""Convert an external wavefunction to a PennyLane state vector.
+
+    Args:
+        solver: external wavefunction object that will be converted
+        method (str): keyword specifying the calculation method
+
+        tol (float): the tolerance to which the wavefunction is being built -- Slater
+            determinants with coefficients smaller than this tolerance are discarded.
+
+    Returns:
+        statevector (array): normalized state vector of length 2**len(number_of_spinorbitals)
```

Comment:
If your function raises an error then we include that in the doc-string as a section: 
```suggestion
    Raises: 
        ValueError: *description of source of error*
    Returns:
        statevector (array): normalized state vector of length 2**len(number_of_spinorbitals)
```